src/main/java/org/elasticsearch/index/mapper/core/TypeParsers.java

```diff
@@ -249,27 +251,20 @@ public static void parseField(AbstractFieldMapper.Builder builder, String name,
             } else if (propName.equals("index_options")) {
                 builder.indexOptions(nodeIndexOptionValue(propNode));
                 iterator.remove();
-            } else if (propName.equals("analyzer")) {
+            } else if (propName.equals("analyzer") ||
+                       propName.equals("index_analyzer") && parserContext.indexVersionCreated().before(Version.V_2_0_0)) {
```

Comment:
Even if the operator order guarantees that the AND will apply before the OR, maybe add parenthesis to make it more explicit?